app/eventyay/presale/templates/pretixpresale/fragment_modals.html

```diff
@@ -14,4 +14,380 @@ <h3></h3>
+            });
+
+            const registerLink = modal.querySelector('.checkout-login-footer a');
+            if (registerLink) {
+                try {
+                    const url = new URL(registerLink.href, window.location.origin);
+                    url.searchParams.set(REDIRECT_PARAM, checkoutUrl);
+                    registerLink.href = url.toString();
+                } catch (e) {
+                    // Fallback if URL constructor fails with invalid URL
+                    // Build URL manually as last resort
+                    registerLink.href = `${registerLink.href.split('?')[0]}?${REDIRECT_PARAM}=${encodeURIComponent(checkoutUrl)}`;
+                }
+            }
+
+            modal.style.display = 'flex';
+        });
+    }
+
+    // Centralized function to close modal and cleanup
+    const closeModal = () => {
+        modal.style.display = 'none';
+        // Clean up SSO popup interval if active
+        if (activeCheckLogin) {
+            clearInterval(activeCheckLogin);
+            activeCheckLogin = null;
+        }
+    };
+
+    // Close modal on close button click
+    if (closeBtn) {
+        closeBtn.addEventListener('click', closeModal);
+    }
+
+    // Close on overlay click
+    modal.addEventListener('click', (e) => {
+        if (e.target === modal) {
+            closeModal();
+        }
+    });
+
+    // Close on Escape key - check if modal is visible using modal.style.display
+    document.addEventListener('keydown', (e) => {
+        if (e.key === 'Escape' && modal.style.display === 'flex') {
+            closeModal();
+        }
+    });
+
+    // AJAX form submission
+    if (loginForm) {
+        loginForm.addEventListener('submit', async (e) => {
+            e.preventDefault();
+
+            const submitBtn = document.getElementById('checkout-login-submit');
+            const originalText = submitBtn.innerHTML;
+            submitBtn.disabled = true;
+            submitBtn.innerHTML = '<i class="fa fa-spinner fa-spin"></i> {% trans "Logging in..." %}';
+
+            if (errorDiv) errorDiv.style.display = 'none';
+
+            const formData = new FormData(loginForm);
+
+            try {
+                const response = await fetch(loginForm.action, {
+                    method: 'POST',
+                    body: formData,
+                    credentials: 'same-origin',
+                    headers: {
+                        'X-Requested-With': 'XMLHttpRequest'
+                    }
+                });
+
+                // Check for successful response (200-299) or redirect (302)
+                // Login endpoint returns 302 on success, not 200
+                if (response.ok || response.status === 302) {
+                    // Login successful, redirect to checkout
+                    window.location.href = checkoutUrl;
+                } else {
+                    throw new Error('{% trans "Invalid email or password. Please try again." %}');
```

Comment:
Just let the original error (`HTTPError`) raise because it has more detail for debugging and also more accurate.

By replacing with `new Error`, you obscure the true error, it may not be "wrong user name or password".